s2n.mk

```diff
@@ -118,7 +118,9 @@ ifdef S2N_ADDRESS_SANITIZER
 endif
 
 ifdef S2N_DEBUG
```

Comment:
Travis and codebuild use S2N_DEBUG with valgrind https://github.com/awslabs/s2n/blob/master/codebuild/bin/s2n_codebuild.sh#L53 . Dropping optimizations for it will likely make test a a lot slower than it is today.